src/buildHTML.js

```diff
@@ -575,12 +579,48 @@ var groupTypes = {
             totalHeight += depth + gap; // \@yargarraycr
             body[r] = outrow;
         }
+
         var offset = totalHeight / 2 + fontMetrics.metrics.axisHeight;
         var coldescriptions = group.value.cols || [];
         var cols = [];
         var colsep;
-        for (c = 0; c < nc; ++c) {
-            var coldescr = coldescriptions[c] || {};
+        var colDescrNum;
+        for (c = 0, colDescrNum = 0;
+             c < nc || colDescrNum < coldescriptions.length;
+             ++c, ++colDescrNum) {
+            var coldescr = coldescriptions[colDescrNum] || {};
+
+            var lastWasSeparator = false;
+            while (coldescr.type === "separator") {
+                if (lastWasSeparator) {
+                    colsep = makeSpan(["arraycolsep"], []);
+                    colsep.style.width = "0.2em";
```

Comment:
I found a quote in http://get-software.net/macros/latex/required/tools/array.pdf that might shed some light on the matter:
> In the standard array/tabular implementation this error is not so noticeable
as that version contains negative spacing so that each | takes up no horizontal
space. But since in this package the vertical lines take up their natural
width one sees two lines if two are specified.

I'm sure if we got the "width" metric for `\vert` it would turn out to be close `0.2em`.